src/core.alp

```diff
@@ -0,0 +1,88 @@
+module core
+
+export (|>), (>>), identity, curry, uncurry
+
+{-| Forward function application `x |> f == f x`. This function is useful
+    for avoiding parentheses and writing code in a more natural way.
+    Consider the following code to create a pentagon:
+
+        scale 2 (move (10,10) (filled blue (ngon 5 30)))
+
+    This can also be written as:
+        ngon 5 30
+          |> filled blue
+          |> move (10,10)
+          |> scale 2
+-}
+val (|>) 'a 'b : fn 'a (fn 'a -> 'b) -> 'b
+let (|>) x f =
+  f x
+
+test "forward pipe" =
+  let add x y = x + y in
+  assert.equal (1 |> add 1 |> add 1) 3
+
+-- TODO: This parses incorrectly.
+-- (╯°□°）╯︵ ┻━┻ {not_exported,<<"equal">>,1}
+-- What is the right way to write this? I think I need to specify the
+-- the associativity of (|>)
+--
+-- test "forward pipe chained into equal" =
+--   let add x y = x + y in
+--    1
+--     |> add 1
+--     |> add 1
+--     |> (assert.equal 3)
+
+
+{-| Function composition, passing results along in the suggested direction. For
+    example, the following code checks if the square root of a number is odd:
+
+        sqrt >> isEven >> not
+-}
+val (>>) 'a 'b 'c : fn (fn 'a -> 'b) (fn 'b -> 'c) -> (fn 'a -> 'c)
+let (>>) f g x =
```

Comment:
Found the problem, `>>` is reserved for binaries (as is `<<`) which is tripping up how the scanner infers breaks between functions (it tracks whether or not it's within a binary expression).

Reply:
Ah yes! Good catch, thank you. What operator would you suggest for forwards composition?

Reply:
I've previously defined these in some code of mine as `>>>` and `<<<`.

Reply:
Works for me